File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.py
from collections import deque

def solution(arr):
    q = deque(arr)
    n = q.popleft()
    answer = [n]
    
    while q:
        v = q.popleft()
        if n == v:
            continue
        answer.append(v)
        n = v
        
    return answer

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;
    answer.reserve(arr.size());

    for (int num: arr){
        if (answer.empty() || answer.back() != num){
            answer.push_back(num);
        }
    }

    return answer;
}